src/yunohost/app.py

```diff
@@ -341,6 +341,105 @@ def app_map(app=None, raw=False, user=None):
     return result
 
 
+def app_changeurl(auth, app, domain, path):
+    """
+    Modify the URL at which an application is installed.
+
+    Keyword argument:
+        app -- Taget app instance name
+        domain -- New app domain on which the application will be moved
+        path -- New path at which the application will be move
+
+    """
+    from yunohost.hook import hook_exec
+
+    installed = _is_installed(app)
+    if not installed:
+        raise MoulinetteError(errno.ENOPKG,
+                              m18n.n('app_not_installed', app=app))
+
+    if not os.path.exists(os.path.join(APPS_SETTING_PATH, app, "scripts", "change_url")):
+        raise MoulinetteError(errno.EINVAL, m18n.n("app_change_no_change_url_script", app_name=app))
+
+    old_domain = app_setting(app, "domain")
+    old_path = app_setting(app, "path")
```

Comment:
Some example: odoo, gitlab and many others aren't designed to be hosted on a subpath. So packager just ask for one question: on which domain.

This kind of apps could run change_url to change the domain only.

More generally, it asks the question of changing install parameters.